src/V2World/Countries/CommonCountriesDataFactory.cpp

```diff
@@ -8,63 +11,36 @@
 namespace
 {
 
-bool ShouldLineBeSkipped(std::string line)
-{
-	line.erase(std::remove_if(line.begin(), line.end(), ::isspace), line.end());
-	return line.starts_with('#') || line.size() < 3 || line.starts_with("dynamic_tags");
-}
-
-
-std::string ExtractCountryFileName(const std::string& country_file_line)
-{
-	const auto start = country_file_line.find_first_of('/');
-	const auto size = country_file_line.find_last_of('\"') - start;
-
-	return country_file_line.substr(start, size);
-}
-
-
 std::tuple<std::map<std::string, Vic2::CommonCountryData>, std::vector<Vic2::Party>> ProcessCountriesDotTxt(
 	 const std::string& country_list_file,
 	 const commonItems::ModFilesystem& mod_filesystem)
 {
 	std::map<std::string, Vic2::CommonCountryData> common_countries_data;
 	std::vector<Vic2::Party> parties;
 
-	const auto possible_file = mod_filesystem.GetActualFileLocation(country_list_file);
-	if (!possible_file)
-	{
-		return std::make_tuple(common_countries_data, parties);
-	}
-
-	std::ifstream vic2_countries_input(*possible_file);
-	if (!vic2_countries_input.is_open())
-	{
-		return std::make_tuple(common_countries_data, parties);
-	}
-
 	Vic2::CommonCountryData::Factory common_country_data_factory;
-	while (!vic2_countries_input.eof())
-	{
-		std::string line;
-		getline(vic2_countries_input, line);
-		if (ShouldLineBeSkipped(line))
-		{
-			continue;
-		}
 
-		auto tag = line.substr(0, 3);
-		auto country_file_name = ExtractCountryFileName(line);
-		auto country_data = common_country_data_factory.ImportCommonCountryData(country_file_name, mod_filesystem);
-		common_countries_data[tag] = *country_data;
-		for (const auto& party: country_data->GetParties())
-		{
-			parties.emplace_back(party);
-		}
+	commonItems::parser countries_parser;
+	countries_parser.registerKeyword("dynamic_tags", commonItems::ignoreItem);
+	countries_parser.registerRegex(commonItems::catchallRegex,
+		 [&common_countries_data, &parties, &common_country_data_factory, mod_filesystem](const std::string& tag,
+			  std::istream& input_stream) {
+			 const std::string country_file_name = commonItems::getString(input_stream);
+			 const std::unique_ptr<Vic2::CommonCountryData> country_data =
+				  common_country_data_factory.ImportCommonCountryData(country_file_name, mod_filesystem);
+			 common_countries_data[tag] = *country_data;
+			 for (const Vic2::Party& party: country_data->GetParties())
+			 {
+				 parties.emplace_back(party);
+			 }
+		 });
+
+	if (const std::optional<std::string> possible_file = mod_filesystem.GetActualFileLocation(country_list_file);
+		 possible_file)
+	{
+		countries_parser.parseFile(*possible_file);
```

Comment:
I'd only construct the parser once we're sure the file exists, so inside this block.